tests/testthat/test-shinytest2-modules.R

```diff
@@ -0,0 +1,117 @@
+testthat::test_that("e2e: the module server logic is only triggered when the teal module becomes active", {
+  value_export_module <- function(label = "custom module") {
+    module(
+      label = label,
+      server = function(id, data) {
+        moduleServer(id, function(input, output, session) {
+          shiny::exportTestValues(
+            value = rnorm(1)
+          )
+        })
+      },
+      ui = function(id) {
+        ns <- NS(id)
+        h1("Module that exports a random value for testing")
+      }
+    )
+  }
+
+  app <- TealAppDriver$new(
+    data = simple_teal_data(),
```

Comment:
Consider this (with additional changes in helper file):
```suggestion
    data = simple_teal_data,
```

In `helper-xyz.R` we can create objects to be used across test files - not only functions.

This should be a little bit faster.

Reply:
Using an object instead of a function could be a problem if the `simple_teal_data` object is modified. 

I know that none of these tests or future ones should change this, but it seems reasonable to have a function. 

Performance-wise it has a very low overhead compared to `shinytest2` runtimes

``` r
microbenchmark::microbenchmark(build = simple_teal_data(), unit = "ms")
#> Unit: milliseconds
#>   expr      min       lq      mean    median        uq    max neval
#>  build 0.506708 0.588549 0.8417622 0.6442315 0.7887155 14.569   100
```

Reply:
Yes - you are definitely right about modification. OK, this is definitely fine to leave it as function. Closing